addons/hw_drivers/tools/helpers.py

```diff
@@ -47,24 +60,35 @@ def add_credential(db_uuid, enterprise_code):
 def check_certificate():
     """
     Check if the current certificate is up to date or not authenticated
+    :return CheckCertificateStatus
     """
     server = get_odoo_server_url()
-    if server:
-        path = Path('/etc/ssl/certs/nginx-cert.crt')
-        if path.exists():
-            with path.open('r') as f:
-                cert = crypto.load_certificate(crypto.FILETYPE_PEM, f.read())
-                cert_end_date = datetime.datetime.strptime(cert.get_notAfter().decode('utf-8'), "%Y%m%d%H%M%SZ") - datetime.timedelta(days=10)
-                for key in cert.get_subject().get_components():
-                    if key[0] == b'CN':
-                        cn = key[1].decode('utf-8')
-                if cn == 'OdooTempIoTBoxCertificate' or datetime.datetime.now() > cert_end_date:
-                    _logger.info(_('Your certificate %s must be updated') % (cn))
-                    load_certificate()
-                else:
-                    _logger.info(_('Your certificate %s is valid until %s') % (cn, cert_end_date))
-        else:
-            load_certificate()
+    if not server:
+        return CheckCertificateStatus(True, False, "ERR_IOT_HTTPS_CHECK_NO_SERVER")
+
+    path = Path('/etc/ssl/certs/nginx-cert.crt')
+    if not path.exists():
+        return CheckCertificateStatus()
+
+    try:
+        with path.open('r') as f:
+            cert = crypto.load_certificate(crypto.FILETYPE_PEM, f.read())
+    except EnvironmentError:
+        _logger.exception("Unable to read certificate file")
+        return CheckCertificateStatus(True, False, "ERR_IOT_HTTPS_CHECK_CERT_READ_EXCEPTION")
```

Comment:
@xmo-odoo 
In fact, it was my plan at the beginning.
At the beginning I was planning to indeed have a foldable "Error details" tab (see picture attached). However, the issue with that is that it's more complicated to edit compared to documentation. For instance, if we find a new case that induce one of this error, it's easier to just add the case in the documentation rather than editing the code (the functional support can do this for instance).

![image](https://user-images.githubusercontent.com/60775325/225943550-6fb6c226-fadc-4deb-8482-9b17f0601df6.png)

Note that it is actually used outside of the warning, it will be shown on the IoT box homepage (with a button next to it that redirect to the documentation (anchored on the error code) ). 
![image](https://user-images.githubusercontent.com/60775325/225944424-474ba7dc-f0b2-430b-90d5-7950195cb231.png)